### DISCUSSION TITLE
OpenAI 推出機器學習工具 Codex，能直接將英文轉成程式碼！

### DISCUSSION THREAD

**Discussion by 32844583** [1101-GS3073-C/Python-Community-GS3073-C#510] [General]
### Codex 能自動寫程式，給工程師更多時間思考問題
Codex 是一個自動寫程式的 AI，旨在給工程師更多的思考時間，而不是花更多時間寫程式。它不僅能夠接受文字指令，還能接受語音指令。Codex 還與微軟合作開發了一個 Word 插件，能夠自動刪除文字中的空格。 Greg Brockman 表示，編程有兩個部分：思考問題並試圖理解它，以及將這些想法映射到現有的程式碼中。 Codex 的目的是讓工程師在第一部分花費更多時間，而不是第二部分。
https://buzzorange.com/techorange/2021/08/12/codex-translate-english-into-code/

**Comment by NcuJerryKao** [comment]
能把文字轉成程式碼 雖有些無法想像 AI 已然道這種程度
但似乎也看到 AI能取代菜鳥工程師的文章
加之這世界的一切本來就很扯蛋 故我相信

**Comment by brian431** [comment]
OpenAI從創立以來真的推出許多令人驚豔的計劃。不禁令人好奇之後還會有什麼有突破性的發展。

**Comment by Ben6uo** [comment]
看來我的程式能力有夠了 我終於等到這一天了

**Comment by williamyen0623** [comment]
原本以為根據虛擬碼自動寫程式的copilot已經夠神了，這個竟然還能用語音輸入

**Comment by BingRong1217** [comment]
節省工程師的時間可以讓他們有更多時間去完成更多事情

**Comment by xCk27x** [comment]
很好用ㄟ
發現測了一些簡單指令回答都有模有樣的
example:
<img width="701" alt="螢幕擷取畫面_20221229_043506" src="https://user-images.githubusercontent.com/90547641/209925749-5300c927-319d-4e0f-89c2-33920c5172a7.png">

**Comment by limitx0** [comment]
有時候寫程式碼真的很無聊又浪費時間
腦袋已經想很遠了但是就算打字再快輸入速度還是會跟不上
能直接將英文轉成程式碼真的幫大忙了

**Comment by Chenszuhong** [comment]
這個主題真是不錯的想法 感覺都很有趣而且很帥 想要知道更多能延生應用的地方

**Comment by Yilin20021016** [comment]
但會不會出很多BUG阿 因為只是為了可以動

**Comment by magicmicheal** [comment]
我認為這是個有用的功能，之前在自學Python明明已經想到到底要做什麼了，但還要花時間去翻程式書看哪個適合還有它的使用方法

**Comment by larry40305** [comment]
我覺得未來會越來越需要會寫程式的人才，因此類似的相關技術應該會有不錯的發展空間

**Comment by HsuChen88** [comment]
OpenAI在各方面總是做出人們意想不到的突破，會不會有一天我也在裡面工作?

**Reply by chia-kai-chang** [reply]
對於在裡面工作，你有沒有什麼計畫或策略呢？

**Comment by ghost** [comment]
對於初學者而言，有時候寫一些程式就需要掉他們不少時間，如果使用這AI，那麼他們就可以透過觀察AI所生成出來的程式碼來使自己進步，縮短學習時間。

**Comment by Bonus502530** [comment]
身為資工系的學生看到Codex的問世，有兩種截然不同的心情出現，一則以喜 一則以憂，文章中提到Codex可以為寫程式的人省去一些重複性高相較簡單的部分，讓省下的時間專注在解法上，這對寫程式的人幫助蠻大的。雖然現在Codex在準確度上還只有28%，但不難保證未來能否取代人類，而且在這一行業競爭也很激烈，若未來沒有學習一些更需要技術性的知識，未來一些相較低階的工作真的有可能會被機器取代
